shell/platform/android/io/flutter/embedding/engine/systemchannels/PlatformChannel.java

```diff
@@ -4,26 +4,536 @@
+      }
+    }
+  };
 
   public PlatformChannel(@NonNull DartExecutor dartExecutor) {
-    this.channel = new MethodChannel(dartExecutor, "flutter/platform", JSONMethodCodec.INSTANCE);
+    channel = new MethodChannel(dartExecutor, "flutter/platform", JSONMethodCodec.INSTANCE);
+    channel.setMethodCallHandler(parsingMethodCallHandler);
+  }
+
+  /**
+   * Sets the {@link PlatformMessageHandler} which receives all events and requests
+   * that are parsed from the underlying platform channel.
+   */
+  public void setPlatformMessageHandler(@Nullable PlatformMessageHandler platformMessageHandler) {
+    this.mPlatformMessageHandler = platformMessageHandler;
+  }
+
+  // TODO(mattcarroll): add support for IntDef annotations, then add @ScreenOrientation
+  private int decodeOrientations(@NonNull JSONArray encodedOrientations) throws JSONException {
+    int requestedOrientation = 0x00;
+    int firstRequestedOrientation = 0x00;
+    for (int index = 0; index < encodedOrientations.length(); index += 1) {
+      String encodedOrientation = encodedOrientations.getString(index);
+      DeviceOrientation orientation = DeviceOrientation.fromValue(encodedOrientation);
+
+      switch (orientation) {
+        case PORTRAIT_UP:
+          requestedOrientation |= 0x01;
+          break;
+        case PORTRAIT_DOWN:
+          requestedOrientation |= 0x04;
+          break;
+        case LANDSCAPE_LEFT:
+          requestedOrientation |= 0x02;
+          break;
+        case LANDSCAPE_RIGHT:
+          requestedOrientation |= 0x08;
+          break;
+      }
+
+      if (firstRequestedOrientation == 0x00) {
+        firstRequestedOrientation = requestedOrientation;
+      }
+    }
+
+    switch (requestedOrientation) {
+      case 0x00:
+        return ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED;
+      case 0x01:
+        return ActivityInfo.SCREEN_ORIENTATION_PORTRAIT;
+      case 0x02:
+        return ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE;
+      case 0x04:
+        return ActivityInfo.SCREEN_ORIENTATION_REVERSE_PORTRAIT;
+      case 0x05:
+        return ActivityInfo.SCREEN_ORIENTATION_USER_PORTRAIT;
+      case 0x08:
+        return ActivityInfo.SCREEN_ORIENTATION_REVERSE_LANDSCAPE;
+      case 0x0a:
+        return ActivityInfo.SCREEN_ORIENTATION_USER_LANDSCAPE;
+      case 0x0b:
+        return ActivityInfo.SCREEN_ORIENTATION_USER;
+      case 0x0f:
+        return ActivityInfo.SCREEN_ORIENTATION_FULL_USER;
+      case 0x03: // portraitUp and landscapeLeft
+      case 0x06: // portraitDown and landscapeLeft
+      case 0x07: // portraitUp, portraitDown, and landscapeLeft
+      case 0x09: // portraitUp and landscapeRight
+      case 0x0c: // portraitDown and landscapeRight
+      case 0x0d: // portraitUp, portraitDown, and landscapeRight
+      case 0x0e: // portraitDown, landscapeLeft, and landscapeRight
+        // Android can't describe these cases, so just default to whatever the first
```

Comment:
subjective: If you split the case statement out into a helper function you wouldn't need to copy it twice.

Reply:
Fair point, but I think in a debugging situation it would actually be easier to read straight through than to think about a recursive call.